### ISSUE TITLE
Docker login & pull extremely slow on WSL2

### ISSUE THREAD

**Issue by ionel-sirbu-crunch** [microsoft/WSL#8733]
### Version

Microsoft Windows [Version 10.0.22000.778]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

5.10.16

### Distro Version

Ubuntu 20.04.4 LTS

### Other Software

Docker Desktop 4.11.1 (84025)

### Repro Steps

I recently moved my development environment from an old desktop to a new Dell Inspiron 14 7420 2-in-1 laptop. I've been using WSL2 & I copied it over using the `export`/`import` WSL tools for that matter. Everything went fine there, but then after installing Docker Desktop on the new machine, I noticed that certain commands (e.g. `docker login` & `docker pull`) take much, much longer when run from Linux side compared to running them from Windows side.

This can't be a matter of volume mounts, since these commands don't involve any volumes anyway. I haven't even tried a `docker build` yet to be honest, as those other 2 commands are more important to me.

Also, I've allocated enough resources to the WSL2 VM via the `.wslconfig` file, so that shouldn't be a problem either:
```
[wsl2]
memory=8GB
swap=0
localhostForwarding=true
processors=4
```

### Expected Behavior

`docker login` & `docker pull` execution times on WSL2 to be similar to the ones from Windows side.

### Actual Behavior

For example, here's a comparison for the `docker login` command:

- Powershell: 4.117 s
- **Linux: 13 m 40.795 s**

### Diagnostic Logs

`strace` log for `docker login`: https://gist.github.com/ionel-sirbu-crunch/7c3a11caee41c4241ed067e88d5cd1a8
[WslLogs-2022-08-17_12-08-55.zip](https://github.com/microsoft/WSL/files/9359107/WslLogs-2022-08-17_12-08-55.zip)

**Comment by ionel-sirbu-crunch** [comment]
Am I missing something in reporting this ticket? It says `1 of 2 tasks` at the top...

**Comment by offroff** [comment]
I have the same problem. Is there anything I can do?

**Comment by Biswa96** [comment]
Can you provide the required times for those commands?

**Comment by offroff** [comment]
docker login: 1 min 55 secs. 1 min 15 secs to see the message "Login succeded", 40 secs to get the prompt back
docker pull alpine : 40 secs

**Comment by Biswa96** [comment]
I can not use `docker login` command because I have no docker registry setup. But here is the time from `docker pull alpine`.

```
$ time docker pull alpine
Using default tag: latest
latest: Pulling from library/alpine
213ec9aee27d: Pull complete
Digest: sha256:bc41182d7ef5ffc53a40b044e725193bc10142a1243f395ee852a8d9730fc2ad
Status: Downloaded newer image for alpine:latest
docker.io/library/alpine:latest

real    0m8.013s
user    0m0.048s
sys     0m0.000s
```

8 seconds, I guess. I am not use Docker Desktop. Just a simple docker installation in Ubuntu 22.04 in WSL2 in Windows 10.

**Comment by offroff** [comment]
I started to look into this problem and noticed my firewall denies UDP requests from the dns server. I simply uninstalled the crap security software I got with my new computer and now everything works as it should! 2 seconds to pull the alpine image.

**Comment by ionel-sirbu-crunch** [comment]
> 

Well, that can't be it for my case though, as I already have both Inbound & Outbound rules allowing traffic into the WSL2 network interface.

**Comment by BriceJamin** [comment]
On the same laptop
- in a virtual machine, docker login is very fast (less than 1 second ?).
- on WSL2 : 

> ➜  time docker login nexus.xxxxx.com:4432 -u=xxxxxxxx -p=xxxxxxxxx
WARNING! Using --password via the CLI is insecure. Use --password-stdin.
Login Succeeded
docker login nexus.xxxxx.com:4432 -u=xxxxxxxx -p=xxxxxxxxx 0.08s user 0.05s system 0% cpu 21.810 total

**Comment by BriceJamin** [comment]
Docker logout is very slow too.
It take ~15sec until the line disapear from $HOME/.dicker/config.json

> ➜  time docker logout nexus.xxxxx.com:4431
Removing login credentials for nexus.xxxxx.com:4431
docker logout nexus.xxxxx.com:4431  0.36s user 0.18s system 0% cpu 1:24.87 total

**Comment by cimabe** [comment]
In wsl removing `"credsStore": "desktop.exe"` from `~/.docker/config.json` temporarily fixed the slow docker commands for me.

But the line gets re-added after a Docker Desktop restart

`~/.docker/config.json`:
```json
{
  "auths": {},
  "credsStore": "desktop.exe"
}
```
Credits: https://github.com/docker/for-win/issues/10247#issuecomment-1126350051 @emiller-tunein

Manually installing `docker-credential-wincred.exe` seems to be a better workaround:
https://github.com/docker/for-win/issues/10247#issuecomment-1152865542

**Comment by ionel-sirbu-crunch** [comment]
> In wsl removing `"credsStore": "desktop.exe"` from `~/.docker/config.json` temporarily fixed the slow docker commands for me.
> 
> But the line gets re-added after a Docker Desktop restart
> 
> `~/.docker/config.json`:
> 
> ```json
> {
>   "auths": {},
>   "credsStore": "desktop.exe"
> }
> ```
> 
> Credits: [docker/for-win#10247 (comment)](https://github.com/docker/for-win/issues/10247#issuecomment-1126350051) @emiller-tunein
> 
> Manually installing `docker-credential-wincred.exe` seems to be a better workaround: [docker/for-win#10247 (comment)](https://github.com/docker/for-win/issues/10247#issuecomment-1152865542)

Thanks for the recommendation @cimabe ! I tried the `docker-credential-wincred.exe` workaround, but unfortunately it didn't change anything for me, it still takes ages for a simple `docker login`. 😭

**Comment by ionel-sirbu-crunch** [comment]
Is there any team moderating these issues? Shall I leave this one open, in case anyone wants to have a look at the logs, or shall I close it, now that I sorted it out with the recommended workaround?

**Comment by cimabe** [comment]
This has been fixed for me in the new `Docker Desktop` `4.16.0` release
https://docs.docker.com/desktop/release-notes/#4160:

>For Windows
> - Fixed docker build hanging while printing “load metadata for”. Fixes [docker/for-win#10247](https://github.com/docker/for-win/issues/10247).
